discovery-infra/test_infra/assisted_service_api.py

```diff
@@ -419,11 +409,8 @@ def create_client(
         url,
         offline_token=utils.get_env('OFFLINE_TOKEN'),
         pull_secret="",
-        wait_for_api=True,
```

Comment:
why not just set it to `False` by default, (except the discovery entry point)?

Reply:
what Raz said. On REST flow we want to fail fast if there's a problem

Reply:
and we might want to keep this logic also for the e2e tests

Reply:
My point here (in the commit message) was that we wait for the service to be ready in the deploy make targets in assisted service, so this isn't really necessary here in any flow.

ref: https://github.com/openshift/assisted-service/blob/89e2797e57d0b77d4fafb151d40cf8fdb6aedd06/Makefile#L227-L231

Additionally this uses the ready endpoint rather than querying for clusters so it wouldn't run into the same auth problem I had here.

In any case it's no longer relevant to this PR so I took the change out.